bundles/org.openhab.core.config.discovery.addon.process/src/main/java/org/openhab/core/config/discovery/addon/process/ProcessAddonFinder.java

```diff
@@ -84,28 +108,26 @@ public Set<AddonInfo> getSuggestedAddons() {
             for (AddonDiscoveryMethod method : candidate.getDiscoveryMethods().stream()
                     .filter(method -> SERVICE_TYPE.equals(method.getServiceType())).toList()) {
 
-                List<AddonMatchProperty> matchProperties = method.getMatchProperties();
-                List<AddonMatchProperty> commands = matchProperties.stream()
-                        .filter(amp -> COMMAND.equals(amp.getName())).toList();
+                Map<String, Pattern> matchProperties = method.getMatchProperties().stream()
+                        .collect(Collectors.toMap(AddonMatchProperty::getName, AddonMatchProperty::getPattern));
 
-                if (matchProperties.size() != commands.size()) {
-                    logger.warn("Add-on '{}' addon.xml file contains unsupported 'match-property'", candidate.getUID());
+                if (matchProperties.isEmpty()) {
+                    logger.warn("Add-on '{}' addon.xml file contains no 'match-property'", candidate.getUID());
+                    break;
                 }
 
-                if (commands.isEmpty()) {
-                    logger.warn("Add-on '{}' addon.xml file does not specify match property \"{}\"", candidate.getUID(),
-                            COMMAND);
+                Set<String> propertyNames = new HashSet<>(matchProperties.keySet());
+                propertyNames.removeAll(SUPPORTED_PROPERTIES);
+                if (!propertyNames.isEmpty()) {
+                    logger.warn("Add-on '{}' addon.xml file contains unsupported 'match-property' [{}]",
+                            candidate.getUID(), String.join(",", propertyNames));
                     break;
```

Comment:
Do you think "break" should be removed here? It may cause backward compatibility issue if we want to add a new match property (for newer add-ons running on an old core version).

At least one match property is there, so we can safely run the check below.

Reply:
Agree, warning and ignoring should be sufficient.

Reply:
> one match property is there, so we can safely run the check below.

If the only match property were an as yet unsupported one, the current code would result in a false positive. But I can fix that.

PS I also have an idea to optimise the command line parsing.

Reply:
At the moment, that is not an issue, as the finders only work on distribution add-ons. And there it is no issue as versions will be compatible. The marketplace add-ons will not be considered in the finders. So this whole discussion is in the end not relevant anyway.